From 62d99e3443be94cd039ad9ea852274ff087c57a5 Mon Sep 17 00:00:00 2001
From: Eli Nidam <elini@marvell.com>
Date: Tue, 29 Apr 2014 13:35:33 +0300
Subject: [PATCH 1598/1825] spi: move SPI write register to HAL

https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
commit c94faafce2a3150f91550de71e49d8ae7531e264

	Create new SPI function mvSpiIfByteLenSet
	Replace write register configuration with HAL function

Change-Id: I0c7ec568d09912d7f75f2e305956220e8be75079
Signed-off-by: Eli Nidam <elini@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/7754
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 arch/arm/plat-armada/mv_hal/spi/mvSpi.c |   26 ++++++++++++++++++++++++++
 arch/arm/plat-armada/mv_hal/spi/mvSpi.h |    4 ++++
 2 files changed, 30 insertions(+), 0 deletions(-)

diff --git a/arch/arm/plat-armada/mv_hal/spi/mvSpi.c b/arch/arm/plat-armada/mv_hal/spi/mvSpi.c
index d9473e6..6f9f88d 100644
--- a/arch/arm/plat-armada/mv_hal/spi/mvSpi.c
+++ b/arch/arm/plat-armada/mv_hal/spi/mvSpi.c
@@ -852,4 +852,30 @@ MV_STATUS mvSpiReadWrite(MV_U8 spiId, MV_U8 *pRxBuff, MV_U8* pTxBuff, MV_U32 buf
 	return MV_OK;
 }
 
+/*******************************************************************************
+* mvSpiIfByteLenSet - Number of bytes in each serial flash I/O transfer.
+*
+* DESCRIPTION:
+*
+* INPUT:
+*       spiId: The SPI controller ID to setup.
+*	byteLen : 0 = 1Byte, 1 = 2Byte
+*
+* OUTPUT:
+*       None.
+*
+* RETURN:
+*
+*******************************************************************************/
+MV_VOID mvSpiIfByteLenSet(MV_U8 spiId, MV_U32 byteLen)
+{
+	MV_U32	ctrlReg;
+
+	ctrlReg = MV_REG_READ(MV_SPI_IF_CONFIG_REG(spiId));
+	ctrlReg &= ~(MV_SPI_BYTE_LENGTH_MASK);
+	if (byteLen == SPI_CONFIG_2_BYTE_LEN)
+		ctrlReg |= MV_SPI_BYTE_LENGTH_MASK;
+
+	MV_REG_WRITE(MV_SPI_IF_CONFIG_REG(spiId), ctrlReg);
+}
 
diff --git a/arch/arm/plat-armada/mv_hal/spi/mvSpi.h b/arch/arm/plat-armada/mv_hal/spi/mvSpi.h
index 66ed1f5..77fb2a7 100644
--- a/arch/arm/plat-armada/mv_hal/spi/mvSpi.h
+++ b/arch/arm/plat-armada/mv_hal/spi/mvSpi.h
@@ -73,6 +73,9 @@ extern "C" {
 #include "ctrlEnv/mvCtrlEnvSpec.h"
 #include "mvSysSpiConfig.h"
 
+#define SPI_CONFIG_1_BYTE_LEN	0
+#define SPI_CONFIG_2_BYTE_LEN	1
+
 typedef struct {
 	MV_U16		ctrlModel;
 	MV_U32		tclk;
@@ -151,6 +154,7 @@ MV_STATUS mvSpiParamsSet(MV_U8 spiId, MV_U8 csId, MV_SPI_TYPE type);
 MV_STATUS mvSpiCsSet(MV_U8 spiId, MV_U8 csId);
 MV_STATUS mvSpiIfConfigSet(MV_U8 spiId, MV_SPI_IF_PARAMS *ifParams);
 MV_STATUS mvSpiTimingParamsSet(MV_U8 spiId, MV_SPI_TIMING_PARAMS *tmngParams);
+MV_VOID mvSpiIfByteLenSet(MV_U8 spiId, MV_U32 byteLen);
 
 #endif
 
-- 
1.7.5.4

